util/topgen/validate.py

```diff
@@ -259,6 +298,99 @@
+reset_requests_optional = {
+    'int': ['s', 'internal request list'],
+    'debug': ['s', 'debug request list'],
+    'peripheral': ['s', 'peripheral request list'],
+}
+reset_requests_added = {}
+
+reset_request_required = {
+    'name': ['s', 'the reset request name'],
+    'desc': ['s', 'the reset request descrption'],
+    'module': ['s', 'the reset request source'],
+}
+reset_request_optional = {
+    'width': ['d', 'TODO'],
+}
+reset_request_added = {}
+
+wakeup_required = {
+    'name': ['s', 'the wakeup name'],
+    'width': ['d', 'the width of the signal'],
+    'module': ['s', 'the module sourcing the wakeup'],
+}
+wakeup_optional = {}
+wakeup_added = {}
+
+alert_required = {
+    'name': ['s', 'name of the alert signal'],
+    'width': ['d', 'the number of alerts in this signal, typically 1'],
+    'async': ['s', 'string interpreted as boolean'],
+    'module_name': ['s', 'The module name of the source'],
+}
+alert_optional = {
+    'desc': ['s', 'the description of the alert'],
+    'lpg_name': ['s', 'the low power group of the alert'],
+    'lpg_idx': ['d', 'the index in the lpg group'],
+    'type': ['s', 'should contain "alert"'],
+}
+alert_added = {}
+
+interrupt_required = {
+    'name': ['s', 'the name of the interrupt'],
+    'width': ['d', 'the number of interrupts in this signal, typically 1'],
+    'module_name': ['s', 'The module name of the source'],
+    'intr_type': ['s', 'The IntrType, either Event or Status'],
+    'default_val': ['s', 'a string interpreted as boolean'],
+    'incoming': ['s', 'a string interpreted as boolean'],
+}
+interrupt_optional = {
+    'desc': ['s', 'the description of the interrupt'],
+    'type': ['s', 'should contain "interrupt"'],
+}
+interrupt_added = {}
+
+param_required = {
+    'name': ['s', 'the parameter name'],
+    'desc': ['s', 'the parameter description'],
+    'type': ['s', 'the data type of the parameter'],
+    'default': ['s', 'the default value of the parameter'],
+}
+param_optional = {
+    'expose': ['s', 'seems redundant TODO'],
+    'local': ['s', 'whether it is a localparam, interpreted as boolean'],
+    'name_top': ['s', 'the name in the top-level'],
+    'randcount': ['d', 'TODO'],
+    'randtype': ['s', 'whether it is for "data" or "perm"issions'],
+    'randwidth': ['d', 'the number of bits'],
+    'unpacked_dimensions': ['s', 'the unpacked dimensions for arrays'],
+}
+param_added = {}
+
+inter_sig_required = {
+    'name': ['s', 'the name of the signal'],
+    'struct': ['s', 'the data type of the signal'],
+    'type': [
+        's',
+        'whether the signal is unidirectional or part of a request-response '
+        'pair'
+    ],
+    'act': ['s', 'whether it is a request (req) or a response (rsp)'],
+    'width': ['d', 'the number of items of the signal for arrays'],
+}
+inter_sig_optional = {
+    'desc': ['s', 'the inter signal description'],
+    'inst_name': ['s', 'the instance this signal connects to'],
+    'index': ['d', 'not clear TODO'],
```

Comment:
Added during elaboration. It's the index of an element from an inter_sig connection array.

For example, from this input...
https://github.com/lowRISC/opentitan/blob/3563d66411f24bd3017385dfb2636db2edaf5668/hw/top_earlgrey/data/top_earlgrey.hjson#L1168-L1170

comes output like...
https://github.com/lowRISC/opentitan/blob/3563d66411f24bd3017385dfb2636db2edaf5668/hw/top_earlgrey/data/autogen/top_earlgrey.gen.hjson#L19262-L19273

Reply:
Added to the documentation, thanks.